### ISSUE TITLE
Opening preferences uses `vim` even though `$EDITOR` is `nvim`

### ISSUE THREAD

**Issue by KevinSjoberg** [kovidgoyal/kitty#580]
I've installed kitty via Homebrew. `kitty --version` result in `kitty 0.10.1 created by Kovid Goyal`. Running `echo $EDITOR` result in `nvim`. Running `echo $-` result in `himBCHs`, i.e., an interactive shell. Pressing `super+,` inside kitty opens a new os window with `vim` open, not `nvim`, as I would expected.

It seems like kitty is using the `EDITOR` environment variable here: https://github.com/kovidgoyal/kitty/blob/master/kitty/constants.py#L19.

I'm not quite sure what's going on. I'd glad to help in any way possible.

Cheers,
Kevin

**Comment by kovidgoyal** [comment]
Or use the new kitty.conf setting I just added: https://github.com/kovidgoyal/kitty/commit/345b3437f48f36fc3b8b547c9c71fceac5e3e769

**Comment by KevinSjoberg** [comment]
@kovidgoyal makes sense. Thanks a lot for giving me the `kitty.conf` setting. Appreciate it! 👍

**Comment by katsika** [comment]
I have set `editor nvim` in my `kitty.conf` but it still opens `vim` on `cmd+,`. Any idea why?

**Comment by Luflosi** [comment]
Maybe `nvim` is not in your systemwide `PATH`.
Find out where the `nvim` executable is with `type nvim`, then change your kitty shell by setting the option in `kitty.conf`:
`shell python3 -c 'import os; print(os.environ["PATH"]); input("Press Enter to quit...")'`
and restart kitty.
WARNING: kitty will become pretty much useless until you remove/uncomment that config option again, so you'll have to edit that file with something else later.
Hopefully the `python3` executable is in your `PATH`. If not, just use the absolute path to it.
This will print the `PATH` as kitty sees it, before any shell is started. Now check if the path you got from `type nvim` (I mean the directory, without the `nvim` part at the end) was printed. If it is then I have no idea why it's not working but if it isn't then use the output from `type nvim` in the `editor` config option.

**Comment by sterpe** [comment]
@katsika 

I was trying to use `ed(1)` on macOS having similar issue. It seems like on macOS (Mojave) `/etc/profile` is not run when `kitty` opens the preferences editor and so even setting as system-wid `$EDITOR` in `/etc/profile` has no effect.  

Verified by modifying `kitty.conf` with `shell python -c 'import os; print(os.environ); input("Press Enter to quit...")'` which shows that there is no `$EDITOR` environmental variable set.

Anyway, here was my work around:

```
# /etc/profile

...

if [ -x /usr/local/bin/rlwrap ]; then
        EDITOR='/usr/local/bin/rlwrap -nc /bin/ed'
else
        EDITOR='/bin/ed'
fi
```

```
# ~/.config/kitty/kitty.conf

...

editor sh -c '. /etc/profile; $EDITOR $0'

...

```

cc: @Luflosi

**Comment by kovidgoyal** [comment]
/etc/profile is sourced only inside shells, not graphical programs like
kitty. You need to set the environment variable system wide. How to do
that varies from macOS version to version. Or just use the editor option in
kitty.conf if convincing macOS to co-operate is too much hassle.

**Comment by Luflosi** [comment]
@sterpe @kovidgoyal just made commit 50e7351c73c08b4fe3937f02fc6b52e66987b673, which should allow you to set the `EDITOR` environment variable in your `/etc/profile` or other rc file and have kitty respect it if you leave the `editor` setting at the default value.

**Comment by sterpe** [comment]
Sounds good.  FWIW, I realized a simpler way on macOS is just to modify `kitty.conf` with 

```
editor sh -cil "$EDITOR $0"
```

Which then tricks kitty to run (as editor) the interactive login-shell with string command of `$EDITOR $0` and these seems similar in effect to 50e7351, if I understand correctly.

**Comment by jaduffus** [comment]
Kevin,

How do I set the tab stop when in bash shell (.vimrc) as in editing yaml (.yml) files. In native bash on rhel and osx I can edit ~/.vimrc and append

autocmd FileType yaml setlocal ai ts=2 sw=2 et

What is the equivalent in kitty?

Thank you!
jd

**Comment by kovidgoyal** [comment]
There are ANSI control codes to set terminal tabstops that work on all
terminals. A bit of googling will find them for you. 
But if you want to change the tabstops when editing filesthe
place to do it is in the editor config, not the terminal config.

On Thu, Sep 05, 2019 at 05:08:05PM -0700, John Duffus wrote:
> Kevin,
> 
> How do I set the tab stop when in bash shell (.vimrc) as in editing yaml (.yml) files. In native bash on rhel and osx I can edit ~/.vimrc and append
> 
> autocmd FileType yaml setlocal ai ts=2 sw=2 et
> 
> What is the equivalent in kitty?
> 
> Thank you!
> jd
> 
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/issues/580#issuecomment-528653975

-- 
_____________________________________

Dr. Kovid Goyal 
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Comment by g6ai** [comment]
@sterpe 
> ```
> editor sh -cil "$EDITOR $0"
> ```
This line does not work when I press cmd+comma: instead of opening a new kitty config window, the window quickly flashes and closes.

**Comment by jan-xyz** [comment]
> @sterpe
> 
> > ```
> > editor sh -cil "$EDITOR $0"
> > ```
> 
> This line does not work when I press cmd+comma: instead of opening a new kitty config window, the window quickly flashes and closes.

What's `$EDITOR` currently set to? I had to provide an absolute path to the binary to fix this.

**Comment by g6ai** [comment]
@jan-xyz I've been setting `$EDITOR` in my rc files as such: https://github.com/g6ai/dotfiles/blob/f4b6ae8ba40f724afef02f6a6749a1463a52256f/shrc#L3-L18

**Comment by ecklf** [comment]
Setting
```shell
editor  /usr/local/bin/nvim
```
works for me on macOS

**Comment by sathishmanohar** [comment]
> ```shell
> editor  /usr/local/bin/nvim
> ```
Worked for me in Ubuntu. `editor ~/apps/nvim.appimage` didn't work however `editor /home/user/apps/nvim.appimage` worked.
